front/src/common/PageNotFound.tsx

```diff
@@ -1,20 +1,33 @@
 import React from 'react';
 import { Link, useNavigate } from 'react-router-dom';
 import { useTranslation } from 'react-i18next';
+import { SerializedError } from '@reduxjs/toolkit';
 import NavBarSNCF from './BootstrapSNCF/NavBarSNCF';
 import logo from '../assets/pictures/views/projects.svg';
 import { ModalProvider } from './BootstrapSNCF/ModalSNCF/ModalProvider';
+import { ApiError } from './api/emptyApi';
 
-export default function PageNotFound() {
+type Props = {
+  error?: ApiError | SerializedError;
+};
+
+export default function PageNotFound({ error }: Props) {
   const { t } = useTranslation(['home/home', 'common/common']);
   const navigate = useNavigate();
-
   return (
     <ModalProvider>
       <NavBarSNCF appName="OSRD" logo={logo} />
       <main className="mastcontainer mastcontainer-no-mastnav d-flex align-items-center justify-content-center vh-100">
         <div className="p-3">
-          <h1>{t('home/home:404')}</h1>
+          {error ? (
+            <>
+              <h1>{t(`home/home:${(error as ApiError).status}`)}</h1>
+              <p>{(error as ApiError).data.message}</p>
```

Comment:
Use the utils function, it handles both ApiError and SerializedError
```suggestion
              <p>extractMessageFromError(error)</p>
```